networks.Rmd

```diff
@@ -0,0 +1,259 @@
+
+ggraph builds on top of tidygraph and ggplot2 to allow a complete and familiar grammar of graphics for network data. Still, it is a little different from most ggplot2 extension packages since it works with another data type that is fundamentally different from tabular data. More so, most network visualizations don't concern themselves with mapping variables to `x` and `y` aesthetics since they are concerned with shoving the network topology more than relations between two variables. In order to show network topology the concept of layouts are employed. Layouts are algorithms that use the network structure to calculate (often arbitrary) `x` and `y` values for each node that can then be used for visualization purpose.
+
+### Setting up the visualization
+
+Whereas a normal ggplot2 plot is initialized with a `ggplot()` call, a ggraph plot is initialized with a `ggraph()` call. The first argument is the data, which can be a tbl\_graph or any object convertible to one. The second argument is a layout function and any further argument will be passed on to that function. The default layout will choose an appropriate layout based on the type of graph you provide, but while it is often a descent starting point you should always take control and explore the different layouts available --- networks are nefarious for their ability to show non-existing or exaggerated relations in some layouts.
+
+#### Specifying a layout
+
+The layout argument can either take a string or a function. If a string is provided, the name will be matched to one of the build in layouts (of which there are many). If a function is provided it is assumed that the function takes a tbl\_graph and returns a data frame with at least an x and y column and with the same number of rows as there are nodes in the input graph. Below we can see examples of using the default layout, specifying a specific layout, and providing a layout function directly:
+
+```{r}
+library(ggraph)
+ggraph(hs_graph) + 
+  geom_edge_link() + 
+  geom_node_point()
+```
+
+```{r}
+ggraph(hs_graph, layout = "drl") + 
+  geom_edge_link() + 
+  geom_node_point()
+```
+
+```{r}
+random_layout <- function(x) {
+  n_nodes <- with_graph(x, graph_order())
+  data.frame(
+    x = runif(n_nodes),
+    y = runif(n_nodes)
+  )
+}
+ggraph(hs_graph, layout = random_layout) + 
+  geom_edge_link() + 
+  geom_node_point()
+```
+
+In order to show the graph above we're using the `geom_edge_link()` and `geom_node_point()` functions, and while we have not yet discussed these the do exactly what you may imagine: drawing nodes as points and edges as straight lines.
+
+#### Circularity
+
+Some layouts may be used in both a linear and circular version. The correct way to change this in ggplot2 would be to use `coord_polar()` to change the coordinate system, but since we only want to change the position of nodes in the layout, and not affect the geoms in any other way, this is a function of the layout. The following can show the difference:
```

Comment:
```suggestion
Some layouts may be used in both a linear and circular version. The correct way to change this in ggplot2 would be to use `coord_polar()` to change the coordinate system, but since we only want to change the position of nodes in the layout, and not affect the edges, this is a function of the layout. The following can show the difference:
```